### ISSUE TITLE
Storage mismatch @testnet's block 843408

### ISSUE THREAD

**Issue by roman-khimov** [nspcc-dev/neo-go#968]
file BlockStorage_900000/dump-block-844000.json: block 843408, changes number mismatch: 1 vs 0
Problematic tx, neo-go:
```json
{
   "jsonrpc" : "2.0",
   "id" : 1,
   "result" : {
      "executions" : [
         {
            "stack" : [
               {
                  "value" : "00",
                  "type" : "ByteArray"
               },
               {
                  "value" : "01",
                  "type" : "ByteArray"
               }
            ],
            "notifications" : [
               {
                  "contract" : "0xc191b3e4030b9105e59c6bb56ec0d1273cd43284",
                  "state" : {
                     "type" : "Array",
                     "value" : [
                        {
                           "type" : "ByteArray",
                           "value" : "6e616d6568617368"
                        },
                        {
                           "type" : "ByteArray",
                           "value" : "9527e2a060129763d8ac47666d7ca4ae7c6166dd8f0ba2dfde20ccf8f856723d"
                        }
                     ]
                  }
               },
               {
                  "state" : {
                     "value" : [
                        {
                           "type" : "ByteArray",
                           "value" : "72657475726e"
                        },
                        {
                           "type" : "ByteArray",
                           "value" : "0000000000000000000000000000000000000000000000000000000000000000"
                        }
                     ],
                     "type" : "Array"
                  },
                  "contract" : "0xc191b3e4030b9105e59c6bb56ec0d1273cd43284"
               },
               {
                  "contract" : "0x171ca20b36c73cb20b10d2804286eb82f6b93069",
                  "state" : {
                     "value" : [
                        {
                           "value" : "e58f96e588b06f776e6572",
                           "type" : "ByteArray"
                        },
                        {
                           "value" : "0000000000000000000000000000000000000000000000000000000000000000",
                           "type" : "ByteArray"
                        }
                     ],
                     "type" : "Array"
                  }
               },
               {
                  "state" : {
                     "type" : "Array",
                     "value" : [
                        {
                           "type" : "ByteArray",
                           "value" : "436865636b5769746e657373e9aa8ce8af81"
                        },
                        {
                           "type" : "ByteArray",
                           "value" : "00"
                        }
                     ]
                  },
                  "contract" : "0x171ca20b36c73cb20b10d2804286eb82f6b93069"
               },
               {
                  "contract" : "0xc191b3e4030b9105e59c6bb56ec0d1273cd43284",
                  "state" : {
                     "value" : [
                        {
                           "type" : "ByteArray",
                           "value" : "6e616d6568617368"
                        },
                        {
                           "type" : "ByteArray",
                           "value" : "9527e2a060129763d8ac47666d7ca4ae7c6166dd8f0ba2dfde20ccf8f856723d"
                        }
                     ],
                     "type" : "Array"
                  }
               },
               {
                  "contract" : "0x171ca20b36c73cb20b10d2804286eb82f6b93069",
                  "state" : {
                     "value" : [
                        {
                           "value" : "e58f96e588b06e616d6568617368",
                           "type" : "ByteArray"
                        },
                        {
                           "value" : "9527e2a060129763d8ac47666d7ca4ae7c6166dd8f0ba2dfde20ccf8f856723d",
                           "type" : "ByteArray"
                        }
                     ],
                     "type" : "Array"
                  }
               },
               {
                  "state" : {
                     "type" : "Array",
                     "value" : [
                        {
                           "value" : "616c74657274e4bfaee694b9",
                           "type" : "ByteArray"
                        },
                        {
                           "type" : "ByteArray",
                           "value" : "01"
                        }
                     ]
                  },
                  "contract" : "0x171ca20b36c73cb20b10d2804286eb82f6b93069"
               }
            ],
            "trigger" : "Application",
            "vmstate" : "HALT",
            "gas_consumed" : "2.263",
            "contract" : "0x1ec5830faf76dddf9218a5a86f7ac58713e1bbe8"
         }
      ],
      "txid" : "0xfc9edff63c6e7823a2182e033bffdff69080cec3b0125d9de59736f2cd89e64c"
   }
}
```
C#:
```json
{
   "jsonrpc" : "2.0",
   "id" : 1,
   "result" : {
      "txid" : "0xfc9edff63c6e7823a2182e033bffdff69080cec3b0125d9de59736f2cd89e64c",
      "executions" : [
         {
            "stack" : [],
            "gas_consumed" : "0.737",
            "trigger" : "Application",
            "vmstate" : "FAULT",
            "contract" : "0x1ec5830faf76dddf9218a5a86f7ac58713e1bbe8",
            "notifications" : [
               {
                  "contract" : "0xc191b3e4030b9105e59c6bb56ec0d1273cd43284",
                  "state" : {
                     "type" : "Array",
                     "value" : [
                        {
                           "type" : "ByteArray",
                           "value" : "6e616d6568617368"
                        },
                        {
                           "type" : "ByteArray",
                           "value" : "9527e2a060129763d8ac47666d7ca4ae7c6166dd8f0ba2dfde20ccf8f856723d"
                        }
                     ]
                  }
               },
               {
                  "contract" : "0xc191b3e4030b9105e59c6bb56ec0d1273cd43284",
                  "state" : {
                     "value" : [
                        {
                           "value" : "72657475726e",
                           "type" : "ByteArray"
                        },
                        {
                           "type" : "ByteArray",
                           "value" : "0000000000000000000000000000000000000000000000000000000000000000"
                        }
                     ],
                     "type" : "Array"
                  }
               },
               {
                  "state" : {
                     "type" : "Array",
                     "value" : [
                        {
                           "type" : "ByteArray",
                           "value" : "e58f96e588b06f776e6572"
                        },
                        {
                           "type" : "ByteArray",
                           "value" : "0000000000000000000000000000000000000000000000000000000000000000"
                        }
                     ]
                  },
                  "contract" : "0x171ca20b36c73cb20b10d2804286eb82f6b93069"
               }
            ]
         }
      ]
   }
}
```

**Comment by roman-khimov** [comment]
Contract:
```
{
   "result" : {
      "hash" : "0x171ca20b36c73cb20b10d2804286eb82f6b93069",
      "script" : "54c56b6c766b00527ac46c766b51527ac4616c766b00c36c766b52527ac46c766b52c3086e616d6568617368876334006c766b52c305717565727987634a006c766b52c305616c746572876360006c766b52c30664656c65746587638e0062b0006c766b51c300c36c766b51c351c36c766b51c352c361527265d3016c766b53527ac462a4006c766b51c300c36c766b51c351c36c766b51c352c36152726596036c766b53527ac4627f006c766b51c300c36c766b51c351c36c766b51c352c36c766b51c353c3615379517955727551727552795279547275527275652a046c766b53527ac46241006c766b51c300c36c766b51c351c36c766b51c352c36152726565056c766b53527ac4621c000a6d61696ee696b9e6b39561658a006c766b53527ac46203006c766b53c3616c756653c56b6c766b00527ac422000000000000000000000000000000000000000000000000000000000000000000006c766b51527ac452c576006c766b00c3c476516c766b51c3c46168124e656f2e52756e74696d652e4e6f74696679616c766b51c36c766b52527ac46203006c766b52c3616c756653c56b6c766b00527ac46101006c766b51527ac452c576006c766b00c3c476516c766b51c3c46168124e656f2e52756e74696d652e4e6f74696679616c766b51c36c766b52527ac46203006c766b52c3616c756653c56b6c766b00527ac46101016c766b51527ac452c576006c766b00c3c476516c766b51c3c46168124e656f2e52756e74696d652e4e6f74696679616c766b51c36c766b52527ac46203006c766b52c3616c756655c56b6c766b00527ac46c766b51527ac46c766b52527ac4200000000000000000000000000000000000000000000000000000000000000000086e616d656861736853c576006c766b00c3c476516c766b51c3c476526c766b52c3c4615272678432d43c27d1c06eb56b9ce505910b03e4b391c16c766b53527ac452c576000ee58f96e588b06e616d6568617368c476516c766b53c3c46168124e656f2e52756e74696d652e4e6f74696679616c766b53c36c766b54527ac46203006c766b54c3616c756656c56b6c766b00527ac46c766b51527ac46c766b52527ac420000000000000000000000000000000000000000000000000000000000000000005717565727953c576006c766b00c3c476516c766b51c3c476526c766b52c3c4615272678432d43c27d1c06eb56b9ce505910b03e4b391c16c766b53527ac452c576000be58f96e588b06f776e6572c476516c766b53c3c46168124e656f2e52756e74696d652e4e6f74696679616c766b53c36168184e656f2e52756e74696d652e436865636b5769746e6573736c766b54527ac46c766b54c36425006112436865636b5769746e657373e9aa8ce8af816165fcfd6c766b55527ac46225006112436865636b5769746e657373e9aa8ce8af81616586fd6c766b55527ac46203006c766b55c3616c756656c56b6c766b00527ac46c766b51527ac46c766b52527ac4616168164e656f2e53746f726167652e476574436f6e746578746c766b00c36c766b51c36c766b52c361527265d4fd617c680f4e656f2e53746f726167652e4765746c766b53527ac46c766b53c3009c6c766b54527ac46c766b54c3641e00610b7175657279e69fa5e8afa2616577fc6c766b55527ac4623a0052c576000461646472c476516c766b53c3c46168124e656f2e52756e74696d652e4e6f74696679616c766b53c36c766b55527ac46203006c766b55c3616c756659c56b6c766b00527ac46c766b51527ac46c766b52527ac46c766b53527ac4616c766b00c36c766b51c36c766b52c361527265d9fd01016c766b54527ac46c766b54c364f800616c766b00c36c766b51c36c766b52c361527265edfc6c766b55527ac46168164e656f2e53746f726167652e476574436f6e746578746c766b55c3617c680f4e656f2e53746f726167652e4765746c766b56527ac46c766b56c300a06c766b57527ac46c766b57c3643a00616168164e656f2e53746f726167652e476574436f6e746578746c766b55c3617c68124e656f2e53746f726167652e44656c65746561616168164e656f2e53746f726167652e476574436f6e746578746c766b55c36c766b53c3615272680f4e656f2e53746f726167652e507574610c616c74657274e4bfaee694b96165c4fb6c766b58527ac4621f00610c616c74657274e4bfaee694b9616554fb6c766b58527ac46203006c766b58c3616c756658c56b6c766b00527ac46c766b51527ac46c766b52527ac4616c766b00c36c766b51c36c766b52c36152726580fc01016c766b53527ac46c766b53c364f900616c766b00c36c766b51c36c766b52c36152726594fb6c766b54527ac46168164e656f2e53746f726167652e476574436f6e746578746c766b54c3617c680f4e656f2e53746f726167652e4765746c766b55527ac46c766b55c300a06c766b56527ac46c766b56c3646300616168164e656f2e53746f726167652e476574436f6e746578746c766b54c3617c68124e656f2e53746f726167652e44656c657465611b64656c657465e588a0e999a4e59cb0e59d80e698afe590a6e7a9ba616595fa6c766b57527ac4625000611b64656c657465e588a0e999a4e59cb0e59d80e698afe590a6e7a9ba616516fa6c766b57527ac4622500611264656c657465e588a0e999a4e989b4e69d836165f4f96c766b57527ac46203006c766b57c3616c7566",
      "name" : "nnsResolver(addr)",
      "author" : "qingmingzi",
      "code_version" : "v0.0.2",
      "version" : 0,
      "returntype" : "ByteArray",
      "parameters" : [
         "String",
         "Array"
      ],
      "description" : "NEO Name Service Resolver(address)",
      "properties" : {
         "is_payable" : false,
         "dynamic_invoke" : false,
         "storage" : true
      },
      "email" : "matrix3345@hotmail.com"
   },
   "jsonrpc" : "2.0",
   "id" : 1
}
```

**Comment by roman-khimov** [comment]
Parameters: ("alter", ["neo2", "qingmingzi", 0, 0])

**Comment by roman-khimov** [comment]
`Neo.Runtime.CheckWitness` call with the following parameter:
```
[
    {
        "type": "ByteArray",
        "value": "0000000000000000000000000000000000000000000000000000000000000000"
    }
]
```
Testnet is fun.

**Comment by roman-khimov** [comment]
Fixed by #971.
